raiden-ts/src/transport/epics/webrtc.ts

```diff
@@ -130,117 +138,101 @@ async function getMatrixIceServers(matrix: MatrixClient): Promise<RTCIceServer[]
-  matrix: MatrixClient,
-  start$: Observable<null>,
-  { callId, peerId, peerAddress }: CallInfo,
-  deps: Pick<RaidenEpicDeps, 'log' | 'latest$' | 'config$'>,
+  action$: Observable<RaidenAction>,
+  info: CallInfo,
+  { log }: Pick<RaidenEpicDeps, 'log'>,
 ) {
-  const { log, config$ } = deps;
-  return config$.pipe(
-    first(),
-    mergeMap((config) =>
-      merge(
-        // when seeing an icecandidate, send it to peer
-        fromEvent<RTCPeerConnectionIceEvent>(connection, 'icecandidate').pipe(
-          pluck('candidate'),
-          delayWhen(() => start$),
-          takeWhile(isntNil),
-          bufferTime(10),
-          filter((candidates) => candidates.length > 0),
-          tap((e) => log.debug('RTC: got candidates', callId, e)),
-          mergeMap((candidates) => {
-            const body: t.TypeOf<typeof RtcCandidates> = {
-              type: RtcEventType.candidates,
-              candidates,
-              call_id: callId,
-            };
-            return waitMemberAndSend$(
-              peerAddress,
-              matrix,
-              'm.room.message',
-              { msgtype: 'm.notice', body: jsonStringify(body) },
-              deps,
-            );
-          }),
-        ),
-        // when receiving candidates from peer, add it locally
-        matrixWebrtcEvents$(matrix, RtcEventType.candidates, peerId, config).pipe(
-          tap((e) => log.debug('RTC: received candidates', callId, e.candidates)),
-          mergeMap((event) => from((event.candidates ?? []) as RTCIceCandidateInit[])),
-          mergeMap((candidate) =>
-            defer(() => connection.addIceCandidate(candidate)).pipe(
-              catchError((err) => {
-                log.warn(
-                  'RTC: error setting candidate, ignoring',
-                  callId,
-                  connection.connectionState,
-                  candidate,
-                  err,
-                );
-                return EMPTY;
-              }),
-            ),
-          ),
-        ),
-      ),
+  return merge(
+    // when receiving candidates from peer, add it locally
+    matrixWebrtcEvents$(action$, RtcEventType.candidates, info.peerAddress, info.callId).pipe(
+      tap((e) => log.debug('RTC: received candidates', info.callId, e.candidates)),
+      mergeMap((event) => from((event.candidates ?? []) as RTCIceCandidateInit[])),
+      mergeMap(async (candidate) => {
+        try {
+          await connection.addIceCandidate(candidate);
+        } catch (err) {
+          log.warn(
+            'RTC: error setting candidate, ignoring',
+            info.callId,
+            connection.connectionState,
+            candidate,
+            err,
+          );
+        }
+      }),
+      ignoreElements(),
+    ),
+    // when seeing an icecandidate, send it to peer
+    fromEvent<RTCPeerConnectionIceEvent>(connection, 'icecandidate').pipe(
+      pluck('candidate'),
+      takeWhile(isntNil),
+      bufferTime(10),
```

Comment:
Is this seconds? Seems quite long especially when setting up the connection...

Reply:
No, this is millis. This isn't intended to slow down the rtc connection establishment, but instead simply batch (synchronous or asynchronous) bursts of candidates gathered or cached together, which then gets sent on the same signaling message. This was already here before, just moved it around.